src/ontology/mondo-ingest.Makefile

```diff
@@ -275,15 +275,15 @@ $(REPORTDIR)/%_excluded_terms_in_mondo_xrefs.tsv $(REPORTDIR)/%_excluded_terms_i
 	$(MAKE) up-to-date-mondo.sssom.tsv
 	python3 $(RELEASEDIR)/src/analysis/problematic_exclusions.py \
 	--mondo-mappings-path $(TMPDIR)/mondo.sssom.tsv \
-	--onto-path $(TMPDIR)/component-download-$*.owl.owl \
+	--onto-path $(COMPONENTSDIR)/$*.owl \
```

Comment:
## Bug fix 2
The ICD11 `component-download` had `prefLabel` and `altLabel`, but not `rdfs:label`, which is what it needed. I changed to the processed component to fix that.

I was honestly surprised that this wasn't a lurker bug affecting other ontologies. I had thought that there might be other ones where we do the `prefLabel` --> `rdfs:label`

I'm not sure why for this script we originally went with `component-download` vs the processed component like we usually do.

Reply:
Makes sense! This seems to be the root cause of the issue, and would make sense to me. Good detective work!

Reply:
I'm trying to understand how changing the makefile to use the component directory instead of tmp for "onto-path" fixes the issue if a problematic exclusion is defined as a curie that exists in mondo.sssom.tsv and the mirror(tmp) file, but not the component file based on this [code section](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/analysis/problematic_exclusions.py#L216-L220). If the "problematic" entity is not in the component file, then how does [querying the file in the component directory](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/analysis/problematic_exclusions.py#L233) to get it's label work?

Also, what happens to these "problematic" xrefs and are these icd11.foundation xrefs actually problematic since theses curies are in the component file but use rdf:Description instead?

Reply:
It was my understanding that this is the solution because it has all the preprocessing in it, in particular, it changes skos:prefLabel to rdfs:label so that it can "find a label"?

Reply:
I cannot verify this (neither negatively nor positively).

Eyeballing the goal, it is not clear to me which ontology is needed. As this target seems to be about "exclusions", it may well _need_ the mirror (but I do not know that).

So my suggestion is:

Review the code (python script), determine what  the code gets from the ontology exactly, and based on that, decide:

1. Should the information be extracted from the component (where all relationships are correctly harmonised)
2. Should the information be extract from the mirror (which includes all classes, including the ones _excluded_ from the component
3. Or, god forbid, the information should be extracted from a _harmonized mirror_ which would be... A bit more work. I can do it though if we get to this determination.

Reply:
Sure, I have been reviewing the script. Based on the initial error reported I would be inclined to change the [SPARQL query](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/analysis/problematic_exclusions.py#L32-L41) to:
```
prefix owl:  <http://www.w3.org/2002/07/owl#>
prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
prefix skos: <http://www.w3.org/2004/02/skos/core#>

select ?term_id ?term_label where {
    VALUES ?term_id { {{values}} }
    ?term_id a owl:Class;
    OPTIONAL { ?term_id rdfs:label ?term_label . }
    OPTIONAL { ?term_id skos:prefLabel ?term_label . }
    FILTER(BOUND(?term_label))
```

so that the term label can be found regardless of whether the mirror uses rdfs:label or skos:prefLabel (which icd11.foundation and ICD10CM use). 

This change also allows the `reports/icd11foundation_excluded_terms_in_mondo_xrefs.tsv` to complete successfully.

Reply:
I agree but this does not answer the question fully which ontology (component or mirror) _should_ be used here. Also, it creates the risk of having two labels, as some ontologies have rdfs:label and skos:prefLabel at the same time..

Reply:
A solution from Nico is [mentioned above](https://github.com/monarch-initiative/mondo-ingest/pull/634/files#r1715508237) and no change in the SPARQL query is needed.

Reply:
Running some tests on the options that have been proposed so far this is what I am seeing.

TEST: Run "reports" make goal as: 
`sh run.sh make reports/icd11foundation_excluded_terms_in_mondo_xrefs.tsv 2>&1 | tee icd11foundation_logs.txt`

Option 1: 
- Change only the make goal "$(COMPONENTSDIR)/icd11foundation.owl" to remove "--trim False"

-> Result: 
The set of problematic excluded terms is now larger.
Fails with "RuntimeError: No results found when trying to get labels for terms."


Option 2:
- Change the make goal "$(COMPONENTSDIR)/icd11foundation.owl" to remove "--trim False"
- Change make goal "$(REPORTDIR)/%_excluded_terms_in_mondo_xrefs.tsv" to use "--onto-path $(COMPONENTSDIR)/$*.owl \"

-> Result: 
The set of problematic excluded terms is now larger.
Fails with RuntimeError: No results found when trying to get labels for terms.


Option 3:
- Change the make goal "$(COMPONENTSDIR)/icd11foundation.owl" to remove "--trim False"
- Change the SPARQL query in the script to account for labels using rdfs:label or skos:prefLabel in the mirror files as:
SELECT ?term_id ?term_label WHERE {
    VALUES ?term_id { {{values}} }
    ?term_id a owl:Class .
    OPTIONAL { ?term_id rdfs:label ?term_label . }
    OPTIONAL { ?term_id skos:prefLabel ?term_label . }
    FILTER(BOUND(?term_label))
}

-> Result: This completes successfully.

I thought this SPARQL query would look for rdfs:label and use that if it's found and only if not found then look for skos:prefLabel, therefore reducing the risk if a class uses two label properties.


---
I am still not clear on Joe's logic to change TMPDIR to COMPONENTSDIR when it looks like a problematic exclusion is defined as a curie that exists in mondo.sssom.tsv _and_ the mirror(tmp) file, but not the component file based on this [code section](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/analysis/problematic_exclusions.py#L216-L220).

Reply:
So much going on here. Good work though both of you.

I ran a test yesterday and got the same results when changing from the mirror to the component, but I agree we should / may have to stick to the mirror.

> Joe's logic to change TMPDIR to COMPONENTSDIR 

@twhetzel The reason I did this is because the component always uses `rdfs:label` whereas the mirror doesn't. I was unsure whether or not `prefLabel` was the only alternative label property that might ever appear in the mirror, either. But I knew that whatever label properties might ever exist now or in the future, we'll always standardize them to `rdfs:label` in the component. So I decided to try this change first and make sure I got good results. I ran script and the results were the same, so I went with this as the first implementation.

Reply:
Adding to @matentzn's list of options, "should we query the mirror, component, or harmonized mirror", I would add:
4. If needed, we could use both the mirror and component for different queries.

Reply:
> Option 3: ... remove `--trim False` ... `rdfs:label` or `skos:prefLabel`...

@twhetzel I like this solution.
The `--trim False` is independent of your query since it applies to the component and not the mirror that you are querying in option 3.
As I mentioned above, I was worried there might be additional label properties now or in the future, but you and Nico may know better than I that this isn't a concern. Also if that does happen, we could just add another `OPTIONAL` clause to your query.

@matentzn thanks for the `--trim False` fix. I wonder if this will affect only "problematic exclusions" or might have effects elsewhere.

Reply:
@joeflack4 what results did you get when you say "got the same results when changing from the mirror to the component"?

Reply:
From: [Overall Review](https://github.com/monarch-initiative/mondo-ingest/pull/634#discussion_r1714622167)
> This is what I expected to see. There was only 1 such case in ICD11, and now it is appearing in the results. Further, no other rows were added or removed for other sources.

Reply:
Thanks for explaining your logic for the change TMPDIR to COMPONENTSDIR. I was trying to stay as true to the original code and and how problematic exclusions were identified so I focused on options to have the helper function that looks up term labels work against the files in TMPDIR. 

I realize I did not test the change to the COMPONENTSDIR _and_  your change in the SPARQL query. After your further explanations of choices and my digging through this code further I see why you proposed this change.

I gather it's about 50/50 to include the changes in this PR vs. Option 3. What are the risks that a term that is a problematic exclusion will not exist in the files in COMPONENTSDIR based on the make goals to create the components files?

The test build for Option 3 (just) successfully completed. Is there a PR for the build from this branch (bugfix-icd11-label)?

Reply:
@twhetzel Let's use your approach.
Sometimes what I like to do is merge something that is currently working, and then improve it with a second PR.
I was considering suggesting that we do that w/ my implementation, but I saw your changes which I think are better, and that they giving good results. And now that your build has passed, I think we should take a look at it and if it looks correct, let's push your commit to this PR and merge that.

I did run a build using my approach, but I closed it in favor of yours:
- #635

I think build PRs should really only have 1 commit. They should be based off of 1 implementation. Your changes will produce different results, and I think it's better to have them in a different PR.

Reply:
Ok, let me push the build I did from Option 3.